sentry-android-core/src/main/java/io/sentry/android/core/PerformanceAndroidEventProcessor.java

```diff
@@ -62,20 +69,27 @@ public SentryEvent process(@NotNull SentryEvent event, @NotNull Hint hint) {
 
     // the app start measurement is only sent once and only if the transaction has
     // the app.start span, which is automatically created by the SDK.
-    if (!sentStartMeasurement && hasAppStartSpan(transaction.getSpans())) {
-      final Long appStartUpInterval = AppStartState.getInstance().getAppStartInterval();
-      // if appStartUpInterval is null, metrics are not ready to be sent
-      if (appStartUpInterval != null) {
+    if (!sentStartMeasurement && hasAppStartSpan(transaction)) {
+      final @NotNull TimeSpan appStartTimeSpan =
+          options.isEnablePerformanceV2()
```

Comment:
same here with the api version check. I guess we have to do it everywhere in the PR where we check for performanceV2 flag and return `getAppStartTimeSpan`